server/pipeline_trigger_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+)
+
+const triggerURL = "http://localhost:8080/trigger"
+const pipelineURL = "https://localhost:8080/pipeline/189"
+
+func TestGetPipelineFormData(t *testing.T) {
+	pipelineTrigger := PipelineTrigger{
+		URL:       triggerURL,
+		Token:     "TOKEN",
+		Reference: "cloud",
+		Variables: map[string]string{
+			"A": "B",
+			"C": "%%BIND_TO_C",
+		},
+	}
+
+	data := getPipelineFormData(&pipelineTrigger, []string{"BIND_TO_C=BIND VALUE"})
```

Comment:
Will `BIND_TO_C=BIND VALUE` arrive as a single arg, or will the slash command actually send [`BIND_TO_C=BIND`, `VALUE`]? In general, do we need to worry about spaces in arguments?

Reply:
Slash command will use space as separator and if any value has space then it will pass incorrect value to to the pipeline. At the moment there is no requirement to support spaces in the pipelines and 3/5 that it is not needed. 

But argument validation will be helpful, since users may forgot to use `=` in the definitions. I will add validation to be safe.

Reply:
Argument validation is added.